reactive/containerd.py

```diff
@@ -248,6 +269,50 @@ def merge_custom_registries(config_directory, custom_registries,
     return registries
 
 
+def invalid_custom_registries(custom_registries):
+    """
+    Validate custom registries from config.
+
+    :param str custom_registries: juju config for custom registries
+    :return: error string for blocked status if condition exists, None otherwise
+    :rtype: Optional[str]
+    """
+    try:
+        registries = _registries_list(custom_registries)
+    except json.JSONDecodeError:
+        return "Failed to decode json string"
+    except ValueError as e:
+        return "custom_registries is not a list"
+
+    required_fields = ['url']
+    str_fields = ['host', 'username', 'password', 'ca_file', 'cert_file', 'key_file']
+    truthy_fields = ['insecure_skip_verify', ]
+    restricted_fields = ['ca', 'cert', 'key']
```

Comment:
I would go further and say that *any* unrecognized field is restricted. No need to call these out explicitly.